web/modules/custom/dpl_event/src/Access/EventSeriesTabAccessCheck.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\dpl_event\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Provides an access checker for the event series tab.
+ *
+ * This access check ensures that the tab for editing event instances
+ * is only available when there are more than one instance in the series.
+ */
+class EventSeriesTabAccessCheck implements AccessInterface {
+
+  /**
+   * The entity type manager service.
+   */
+  public function __construct(
+    private EntityTypeManagerInterface $entityTypeManager
+  ) {}
+
+  /**
+   * Access callback for the event series edit instances tab.
+   *
+   * Checks if there are more than one instance in the event series and grants
+   * or denies access based on this condition.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route
+   *   The route match interface to get route parameters.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result, allowing access if there are more than one instance,
+   *   otherwise forbidding it.
+   */
+  public function access(RouteMatchInterface $route): AccessResult {
+    $eventSeriesId = $route->getParameter('eventseries');
+    if (!$eventSeriesId) {
+      return AccessResult::forbidden();
+    }
+
+    $eventSeriesEntity = $this->entityTypeManager->getStorage('eventseries')->load($eventSeriesId);
+    if (!$eventSeriesEntity || !method_exists($eventSeriesEntity, 'getInstanceCount')) {
+      return AccessResult::forbidden();
+    }
+
+    $instanceCount = $eventSeriesEntity->getInstanceCount();
+    return AccessResult::allowedIf($instanceCount > 1)
+      ->addCacheableDependency($eventSeriesEntity)
+      ->cachePerPermissions();
```

Comment:
About the caching problem we saw earlier: I think the problem may be that creating an event instance does not clear the cache for the related event series.

This may also explain the problems we have for displaying the number of instances for each event.

This will probably require some help from either @rasben or myself.

Reply:
This should be looked at together.